src/variant.rs

```diff
@@ -257,13 +272,62 @@ impl Variant {
                     .collect::<Result<Vec<_>, WMIError>>()?;
 
                 Variant::Array(converted_variants)
-            }
+            },
+            Variant::Unknown(u) => {
+                if cim_type == wbemcli::CIM_OBJECT {
+                    Variant::Object(u.to_wbem_class_obj()?)
+                } else {
+                    return Err(WMIError::ConvertVariantError(format!(
+                        "A unknown Variant cannot be turned into a CIMTYPE {}",
+                        &cim_type,
+                    )))
+                }
+            },
+            Variant::Object(o) => Variant::Object(o),
         };
 
         Ok(converted_variant)
     }
 }
 
+/// A wrapper around the [`IUnknown`] interface. \
+/// Used to retrive [`IWbemClassObject`][winapi::um::wbemcli::IWbemClassObject]
+///
+#[derive(Debug, PartialEq, Eq)]
+pub struct IUnknownWrapper {
+    inner: NonNull<IUnknown>
+}
+
+impl IUnknownWrapper {
+    pub unsafe fn new(ptr: NonNull<IUnknown>) -> Self {
+        IUnknownWrapper { inner: ptr }
+    }
+
+    pub fn to_wbem_class_obj(&self) -> WMIResult<IWbemClassWrapper> {
+        unsafe {
+            let ptr = self.inner.as_ptr();
+            let mut obj_ptr = NULL as *mut c_void;
+
+            check_hres((*ptr).QueryInterface(
+                &IID_IWbemClassObject,
+                &mut obj_ptr,
+            ))?;
+
+            let obj = NonNull::new(obj_ptr as *mut _).ok_or(WMIError::NullPointerResult)?;
+            Ok(IWbemClassWrapper::new(obj))
+        }
+    }
+}
+
+impl Serialize for IUnknownWrapper {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer
+    {
+        serializer.serialize_none()
```

Comment:
Document here that this is because it should have been converted to `Variant::Object`